src/main.js

```diff
@@ -0,0 +1,57 @@
+import { createApp } from "vue";
+import App from "./App.vue";
+
+import "bulma/css/bulma.css";
+import "@fortawesome/fontawesome-free/css/all.css";
+import { createStore } from "vuex";
+
+const store = createStore({
+  state() {
+    return {
+      inSidebar: "",
+      postList: [],
+      currentPost: null,
+      userId: null,
+    };
+  },
+
+  mutations: {
+    addPostList(state, posts) {
+      if (Array.isArray(posts)) {
+        state.postList = [...state.postList, ...posts];
+      } else {
+        state.postList.push(posts);
+      }
+    },
+
+    updatePost(state, newPost) {
+      const index = state.postList.findIndex(
+        (post) => post.id === newPost.id
+      );
+
+      state.postList.splice(index, 1, newPost);
+    },
```

Comment:
In the `updatePost` mutation, you're not checking if the post exists in the list before replacing it. If `findIndex` doesn't find the post, it returns `-1`, and `splice` will then remove the last item in the array. To prevent this, you should check if the post exists before updating it.